rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -503,20 +514,30 @@ void debugger_frame::UpdateUnitList()
 
 void debugger_frame::OnSelectUnit()
 {
-	if (m_choice_units->count() < 1) return;
+	if (m_choice_units->count() < 1)
+	{
+		m_debugger_list->UpdateCPUData(nullptr, nullptr);
+		m_breakpoint_list->UpdateCPUData(nullptr, nullptr);
+		m_disasm.reset();
+		cpu.reset();
+		return;
+	}
 
 	const auto weak = m_choice_units->currentData().value<std::weak_ptr<cpu_thread>>();
 	const auto render = m_choice_units->currentData().value<rsx::thread*>();
 
-	if (!render && !weak.owner_before(cpu) && !cpu.owner_before(weak))
+	if (m_emu_state != system_state::stopped)
 	{
-		// They match, nothing to do.
-		return;
-	}
+		if (!render && !weak.owner_before(cpu) && !cpu.owner_before(weak))
+		{
+			// They match, nothing to do.
+			return;
+		}
 
-	if (render && render == this->get_rsx())
-	{
-		return;
+		if (render && render == this->get_cpu())
```

Comment:
Why this-> ? Why not keep it consistent?

Reply:
Removed this->